lib/mumuki/laboratory/mumukit/randomizations/with_randomizations.rb

```diff
@@ -0,0 +1,36 @@
+module WithRandomizations
+  extend ActiveSupport::Concern
+
+  included do
+    serialize :randomizations, Hash
+
+    include InstanceMethods
+    extend ClassMethods
+  end
+
+  module InstanceMethods
+    def seed
+      @seed || 0
+    end
+
+    def seed_with!(seed)
+      @seed = seed
+    end
+  end
+
+  module ClassMethods
+    def randomize(*selectors)
+      selectors.each { |selector| randomize_field selector }
+    end
+
+    private
+
+    def randomize_field(selector)
+      define_method(selector) do |*args|
+        return unless super(*args)
+        interpolations = Mumukit::Randomizer.parse(randomizations).with_seed(seed)
```

Comment:
Please move this two lines to a `randomizer.randomize!(text)` method. E.g.: 

```
Mumukit::Randomizer.parse(randomizations).randomize!(super(*args))
```

Also, in order to avoid reparsing, I'd put a lazily loaded `randomizer`  property on the `WithRandomizations` mixin: 

```
def randomizer
   @randomizer ||= Mumukit::Randomizer.parse(randomizations)
end
```

That way, you don't convert the Json into objects each time you call any of the randomized methods

Reply:
> Please move this two lines to a randomizer.randomize!(text) method. E.g.:

Not sure I got the two lines you meant right. Moving lines 30 and 31 didn't seem to make that much sense, since I'd still need to check if the result was nil before interpolating.

Anyways, I'll fix that wip commit and merge tomorrow o/